rdflib/plugins/sparql/aggregates.py

```diff
@@ -98,10 +100,14 @@ def agg_Count(a, group, bindings):
 
 
 def agg_Sample(a, group, bindings):
-    try:
-        bindings[a.res] = _eval(a.vars, iter(group).next())
-    except StopIteration:
-        pass  # no res
+    for ctx in group:
+        val = _eval(a.vars, ctx)
+        if type(val) is not NotBoundError:
+            bindings[a.res] = val
+            break
+    else:
+        bindings[a.res] = Literal(0)
```

Comment:
Returning a `Literal(0)` where we can't find anything seems wrong to me. Could we return `None` instead?

http://lod.openlinksw.com/sparql?default-graph-uri=&qtxt=SELECT+%3Fx+%28SAMPLE%28%3Fy%29+as+%3Fys%29+%28SAMPLE%28%3Fz%29+as+%3Fzs%29+WHERE+{%0D%0A++++VALUES+%28%3Fx+%3Fy+%3Fz%29+{%0D%0A+++++%282+6+UNDEF%29%0D%0A+++++%282+UNDEF+10%29%0D%0A+++++%283+UNDEF+15%29%0D%0A+++++%283+9+UNDEF%29%0D%0A+++++%285+UNDEF+25%29%0D%0A++++}%0D%0A++}%0D%0A++GROUP+BY+%3Fx&should-sponge=&format=text%2Fhtml&CXML_redir_for_subjs=121&CXML_redir_for_hrefs=&timeout=30000&debug=on